{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XUZS4GUkHIEG"
      },
      "source": [
        "# Подготовка файлов и программ"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "id": "lrtRbP3fAvj8",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "82980332-5f4d-43f2-e6b0-7f6e08f9dbbf"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "\n",
            "# All requested packages already installed.\n",
            "\n",
            "Requirement already satisfied: macs2 in /usr/local/lib/python3.7/site-packages (2.2.7.1)\n",
            "Requirement already satisfied: intervene in /usr/local/lib/python3.7/site-packages (0.6.5)\n",
            "Requirement already satisfied: matplotlib in /usr/local/lib/python3.7/site-packages (from intervene) (3.5.1)\n",
            "Requirement already satisfied: seaborn in /usr/local/lib/python3.7/site-packages (from intervene) (0.11.2)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.7/site-packages (from intervene) (1.3.5)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.7/site-packages (from intervene) (1.21.5)\n",
            "Requirement already satisfied: pybedtools in /usr/local/lib/python3.7/site-packages (from intervene) (0.9.0)\n",
            "Requirement already satisfied: scipy in /usr/local/lib/python3.7/site-packages (from intervene) (1.7.3)\n",
            "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.7/site-packages (from matplotlib->intervene) (1.3.2)\n",
            "Requirement already satisfied: pyparsing>=2.2.1 in /usr/local/lib/python3.7/site-packages (from matplotlib->intervene) (3.0.7)\n",
            "Requirement already satisfied: pillow>=6.2.0 in /usr/local/lib/python3.7/site-packages (from matplotlib->intervene) (9.0.1)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.7/site-packages (from matplotlib->intervene) (0.11.0)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.7/site-packages (from matplotlib->intervene) (21.3)\n",
            "Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.7/site-packages (from matplotlib->intervene) (4.30.0)\n",
            "Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.7/site-packages (from matplotlib->intervene) (2.8.2)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.7/site-packages (from python-dateutil>=2.7->matplotlib->intervene) (1.15.0)\n",
            "Requirement already satisfied: pytz>=2017.3 in /usr/local/lib/python3.7/site-packages (from pandas->intervene) (2021.3)\n",
            "Requirement already satisfied: pysam in /usr/local/lib/python3.7/site-packages (from pybedtools->intervene) (0.18.0)\n"
          ]
        }
      ],
      "source": [
        "!pip install -q condacolab\n",
        "!conda install -c bioconda seqtk\n",
        "!pip install macs2 intervene"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get install bedtools\n",
        "!apt-get install bowtie2 samtools\n",
        "!conda install -c bioconda bowtie2"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dp9zx8Q6MmZD",
        "outputId": "2842dc7a-095d-4ca4-89aa-e17169cc40a0"
      },
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "bedtools is already the newest version (2.26.0+dfsg-5).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 39 not upgraded.\n",
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "bowtie2 is already the newest version (2.3.4.1-1).\n",
            "samtools is already the newest version (1.7-1).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 39 not upgraded.\n",
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "# All requested packages already installed.\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import condacolab\n",
        "condacolab.install()\n",
        "!conda install -c bioconda trimmomatic"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "uuhqn22-MpnO",
        "outputId": "32ad6fc7-ed4a-4a74-ab19-a09fcabbc434"
      },
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "✨🍰✨ Everything looks OK!\n",
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "\n",
            "# All requested packages already installed.\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!java -version\n",
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
        "!unzip fastqc_v0.11.9.zip\n",
        "!chmod a+x FastQC/fastqc\n",
        "# !./FastQC/fastqc --help"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "MkhaycWHMs4A",
        "outputId": "0c227aa8-349d-4238-eff3-62eda6d23e2c"
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "openjdk version \"1.8.0_312\"\n",
            "OpenJDK Runtime Environment (Zulu 8.58.0.13-CA-linux64) (build 1.8.0_312-b07)\n",
            "OpenJDK 64-Bit Server VM (Zulu 8.58.0.13-CA-linux64) (build 25.312-b07, mixed mode)\n",
            "--2022-03-10 18:59:25--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: ‘fastqc_v0.11.9.zip’\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[===================>]   9.77M  1.53MB/s    in 6.4s    \n",
            "\n",
            "2022-03-10 18:59:33 (1.53 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
            "\n",
            "Archive:  fastqc_v0.11.9.zip\n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "  inflating: FastQC/LICENSE          \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/sam-1.103.jar    \n",
            "   creating: FastQC/Templates/\n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "10Fs3qrmHmJi"
      },
      "source": [
        "# Выравнивание"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-z4GQiMfU1wf"
      },
      "source": [
        "## Скачиваем чтения"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "id": "lFWnMg0uU0sJ",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "fd3a128b-2cbd-45f9-a927-c678f1b06850"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-10 18:31:25--  https://www.encodeproject.org/files/ENCFF891GXF/@@download/ENCFF891GXF.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2016/05/30/0ba2fe2d-20a4-4c05-9605-ba30a817c74a/ENCFF891GXF.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF891GXF.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX67YIJFZR&Signature=4%2Bsqzb4afbrsURYVGiNOF%2BoWcZ4%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIr%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJGMEQCICzwbKESX2SuXYyDX%2BdCgEHD%2FB9f5AxE8lB0i7ojli1rAiAbrLuIUIolnIbd1%2FhJx1hGi8cKq1ec8V1RA6GQJXqJaSqDBAjz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMKc9cz76b1MoXw2abKtcDIt1PTzDTZ2cd41%2F6yhBhw7VoZnEJiAxRS3BgVKyIp3mesBtTfMdnhzWtTFz0AHyz9%2BMKJNj3G3fynFEi4TDQTCBqNDR6ag4ttaXiSgPDxfKgvvnQ5fxkx%2FJCt4hP%2FWCxiRkAFcM5amFuaA%2FNxlCpI5oy0yEGpP9o4K3%2B8vKdEl%2FPTrDUn0qFe%2FcDsvS%2Bh7g6aZr5c0Y8O9AsX51xppS8kE9Qma1DAZ02RamsO6DPs8LpZQXHoffZM7idcWjJZlR21mhVTBJLVTDxiYjeju8c%2BffrivTwIkazx%2FauJ%2FgNzOJWtT6FJagEkFYiOPFoHpe%2Fd%2FXacgF9%2FEpA6gmoYeKIMxEaal9SPXT7YmOUylYt5ib7mJV6pmtKoSJz7YVksnIUrqNY4vhFqUH9mHH8FVSBaahyD5VtgGbEPA07W4EKwvBqkI9usqnFrDvyMwdFqIB8YTzGa%2BV3PXyg%2BbWhDqX80hd1mOQ2tYXuYy8xGOuzSZaDPnZf7N19JS8a8b%2BeeX7tyes5NXv%2BnngWNjdnZfaV2CADE1rR2y%2FoGd6kQdY6akuylxgOqrZAc9dmjBQzoZzhSzsWF07mSctgK2yygWd9iQsNp3sYZX65A01OFua%2BjPi7RN%2FoQpNeMNT8qJEGOqYBtbVn%2F9qtb2oaAQCFCScPYA8oBPDduCv%2FqRNABDMrsUQlRgUWrD%2FIdKr3Jgfck%2BwZQQo0ZqcQ0lgoKdDeJdumD2wllUDelGHC5aV9kM7YdDLR8hkxYXBLgaxxXuB7dRcAh%2BrAv8btu3PPNAuNre3Lk0Dyudahd2Np%2Fhbhtd0ZltbH1dVQaX6cFRsn3zV7sYIwwl2X8m0GOQgeDC5PdfQCDFWXvFf4cg%3D%3D&Expires=1647066685 [following]\n",
            "--2022-03-10 18:31:25--  https://encode-public.s3.amazonaws.com/2016/05/30/0ba2fe2d-20a4-4c05-9605-ba30a817c74a/ENCFF891GXF.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF891GXF.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX67YIJFZR&Signature=4%2Bsqzb4afbrsURYVGiNOF%2BoWcZ4%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIr%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJGMEQCICzwbKESX2SuXYyDX%2BdCgEHD%2FB9f5AxE8lB0i7ojli1rAiAbrLuIUIolnIbd1%2FhJx1hGi8cKq1ec8V1RA6GQJXqJaSqDBAjz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMKc9cz76b1MoXw2abKtcDIt1PTzDTZ2cd41%2F6yhBhw7VoZnEJiAxRS3BgVKyIp3mesBtTfMdnhzWtTFz0AHyz9%2BMKJNj3G3fynFEi4TDQTCBqNDR6ag4ttaXiSgPDxfKgvvnQ5fxkx%2FJCt4hP%2FWCxiRkAFcM5amFuaA%2FNxlCpI5oy0yEGpP9o4K3%2B8vKdEl%2FPTrDUn0qFe%2FcDsvS%2Bh7g6aZr5c0Y8O9AsX51xppS8kE9Qma1DAZ02RamsO6DPs8LpZQXHoffZM7idcWjJZlR21mhVTBJLVTDxiYjeju8c%2BffrivTwIkazx%2FauJ%2FgNzOJWtT6FJagEkFYiOPFoHpe%2Fd%2FXacgF9%2FEpA6gmoYeKIMxEaal9SPXT7YmOUylYt5ib7mJV6pmtKoSJz7YVksnIUrqNY4vhFqUH9mHH8FVSBaahyD5VtgGbEPA07W4EKwvBqkI9usqnFrDvyMwdFqIB8YTzGa%2BV3PXyg%2BbWhDqX80hd1mOQ2tYXuYy8xGOuzSZaDPnZf7N19JS8a8b%2BeeX7tyes5NXv%2BnngWNjdnZfaV2CADE1rR2y%2FoGd6kQdY6akuylxgOqrZAc9dmjBQzoZzhSzsWF07mSctgK2yygWd9iQsNp3sYZX65A01OFua%2BjPi7RN%2FoQpNeMNT8qJEGOqYBtbVn%2F9qtb2oaAQCFCScPYA8oBPDduCv%2FqRNABDMrsUQlRgUWrD%2FIdKr3Jgfck%2BwZQQo0ZqcQ0lgoKdDeJdumD2wllUDelGHC5aV9kM7YdDLR8hkxYXBLgaxxXuB7dRcAh%2BrAv8btu3PPNAuNre3Lk0Dyudahd2Np%2Fhbhtd0ZltbH1dVQaX6cFRsn3zV7sYIwwl2X8m0GOQgeDC5PdfQCDFWXvFf4cg%3D%3D&Expires=1647066685\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.133.1\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.133.1|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1287914284 (1.2G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF891GXF.fastq.gz’\n",
            "\n",
            "ENCFF891GXF.fastq.g 100%[===================>]   1.20G  8.50MB/s    in 2m 49s  \n",
            "\n",
            "2022-03-10 18:34:15 (7.25 MB/s) - ‘ENCFF891GXF.fastq.gz’ saved [1287914284/1287914284]\n",
            "\n",
            "--2022-03-10 18:34:15--  https://www.encodeproject.org/files/ENCFF170YTR/@@download/ENCFF170YTR.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2016/05/30/cb086891-b3a0-4a72-a6df-9105d0836dc6/ENCFF170YTR.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF170YTR.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX67YIJFZR&Signature=wUIUex8oSqnoskNlY8jZzvXKPh4%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIr%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJGMEQCICzwbKESX2SuXYyDX%2BdCgEHD%2FB9f5AxE8lB0i7ojli1rAiAbrLuIUIolnIbd1%2FhJx1hGi8cKq1ec8V1RA6GQJXqJaSqDBAjz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMKc9cz76b1MoXw2abKtcDIt1PTzDTZ2cd41%2F6yhBhw7VoZnEJiAxRS3BgVKyIp3mesBtTfMdnhzWtTFz0AHyz9%2BMKJNj3G3fynFEi4TDQTCBqNDR6ag4ttaXiSgPDxfKgvvnQ5fxkx%2FJCt4hP%2FWCxiRkAFcM5amFuaA%2FNxlCpI5oy0yEGpP9o4K3%2B8vKdEl%2FPTrDUn0qFe%2FcDsvS%2Bh7g6aZr5c0Y8O9AsX51xppS8kE9Qma1DAZ02RamsO6DPs8LpZQXHoffZM7idcWjJZlR21mhVTBJLVTDxiYjeju8c%2BffrivTwIkazx%2FauJ%2FgNzOJWtT6FJagEkFYiOPFoHpe%2Fd%2FXacgF9%2FEpA6gmoYeKIMxEaal9SPXT7YmOUylYt5ib7mJV6pmtKoSJz7YVksnIUrqNY4vhFqUH9mHH8FVSBaahyD5VtgGbEPA07W4EKwvBqkI9usqnFrDvyMwdFqIB8YTzGa%2BV3PXyg%2BbWhDqX80hd1mOQ2tYXuYy8xGOuzSZaDPnZf7N19JS8a8b%2BeeX7tyes5NXv%2BnngWNjdnZfaV2CADE1rR2y%2FoGd6kQdY6akuylxgOqrZAc9dmjBQzoZzhSzsWF07mSctgK2yygWd9iQsNp3sYZX65A01OFua%2BjPi7RN%2FoQpNeMNT8qJEGOqYBtbVn%2F9qtb2oaAQCFCScPYA8oBPDduCv%2FqRNABDMrsUQlRgUWrD%2FIdKr3Jgfck%2BwZQQo0ZqcQ0lgoKdDeJdumD2wllUDelGHC5aV9kM7YdDLR8hkxYXBLgaxxXuB7dRcAh%2BrAv8btu3PPNAuNre3Lk0Dyudahd2Np%2Fhbhtd0ZltbH1dVQaX6cFRsn3zV7sYIwwl2X8m0GOQgeDC5PdfQCDFWXvFf4cg%3D%3D&Expires=1647066856 [following]\n",
            "--2022-03-10 18:34:16--  https://encode-public.s3.amazonaws.com/2016/05/30/cb086891-b3a0-4a72-a6df-9105d0836dc6/ENCFF170YTR.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF170YTR.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX67YIJFZR&Signature=wUIUex8oSqnoskNlY8jZzvXKPh4%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIr%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJGMEQCICzwbKESX2SuXYyDX%2BdCgEHD%2FB9f5AxE8lB0i7ojli1rAiAbrLuIUIolnIbd1%2FhJx1hGi8cKq1ec8V1RA6GQJXqJaSqDBAjz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMKc9cz76b1MoXw2abKtcDIt1PTzDTZ2cd41%2F6yhBhw7VoZnEJiAxRS3BgVKyIp3mesBtTfMdnhzWtTFz0AHyz9%2BMKJNj3G3fynFEi4TDQTCBqNDR6ag4ttaXiSgPDxfKgvvnQ5fxkx%2FJCt4hP%2FWCxiRkAFcM5amFuaA%2FNxlCpI5oy0yEGpP9o4K3%2B8vKdEl%2FPTrDUn0qFe%2FcDsvS%2Bh7g6aZr5c0Y8O9AsX51xppS8kE9Qma1DAZ02RamsO6DPs8LpZQXHoffZM7idcWjJZlR21mhVTBJLVTDxiYjeju8c%2BffrivTwIkazx%2FauJ%2FgNzOJWtT6FJagEkFYiOPFoHpe%2Fd%2FXacgF9%2FEpA6gmoYeKIMxEaal9SPXT7YmOUylYt5ib7mJV6pmtKoSJz7YVksnIUrqNY4vhFqUH9mHH8FVSBaahyD5VtgGbEPA07W4EKwvBqkI9usqnFrDvyMwdFqIB8YTzGa%2BV3PXyg%2BbWhDqX80hd1mOQ2tYXuYy8xGOuzSZaDPnZf7N19JS8a8b%2BeeX7tyes5NXv%2BnngWNjdnZfaV2CADE1rR2y%2FoGd6kQdY6akuylxgOqrZAc9dmjBQzoZzhSzsWF07mSctgK2yygWd9iQsNp3sYZX65A01OFua%2BjPi7RN%2FoQpNeMNT8qJEGOqYBtbVn%2F9qtb2oaAQCFCScPYA8oBPDduCv%2FqRNABDMrsUQlRgUWrD%2FIdKr3Jgfck%2BwZQQo0ZqcQ0lgoKdDeJdumD2wllUDelGHC5aV9kM7YdDLR8hkxYXBLgaxxXuB7dRcAh%2BrAv8btu3PPNAuNre3Lk0Dyudahd2Np%2Fhbhtd0ZltbH1dVQaX6cFRsn3zV7sYIwwl2X8m0GOQgeDC5PdfQCDFWXvFf4cg%3D%3D&Expires=1647066856\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.222.35\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.222.35|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1266795314 (1.2G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF170YTR.fastq.gz’\n",
            "\n",
            "ENCFF170YTR.fastq.g 100%[===================>]   1.18G  40.9MB/s    in 30s     \n",
            "\n",
            "2022-03-10 18:34:46 (40.6 MB/s) - ‘ENCFF170YTR.fastq.gz’ saved [1266795314/1266795314]\n",
            "\n",
            "--2022-03-10 18:34:46--  https://www.encodeproject.org/files/ENCFF524OKJ/@@download/ENCFF524OKJ.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2016/05/30/e3880aa9-e9ba-4114-8ba1-f1cf981147f4/ENCFF524OKJ.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF524OKJ.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX67YIJFZR&Signature=j16PNxjyITC%2FHr3WoGJlmxXRv18%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIr%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJGMEQCICzwbKESX2SuXYyDX%2BdCgEHD%2FB9f5AxE8lB0i7ojli1rAiAbrLuIUIolnIbd1%2FhJx1hGi8cKq1ec8V1RA6GQJXqJaSqDBAjz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMKc9cz76b1MoXw2abKtcDIt1PTzDTZ2cd41%2F6yhBhw7VoZnEJiAxRS3BgVKyIp3mesBtTfMdnhzWtTFz0AHyz9%2BMKJNj3G3fynFEi4TDQTCBqNDR6ag4ttaXiSgPDxfKgvvnQ5fxkx%2FJCt4hP%2FWCxiRkAFcM5amFuaA%2FNxlCpI5oy0yEGpP9o4K3%2B8vKdEl%2FPTrDUn0qFe%2FcDsvS%2Bh7g6aZr5c0Y8O9AsX51xppS8kE9Qma1DAZ02RamsO6DPs8LpZQXHoffZM7idcWjJZlR21mhVTBJLVTDxiYjeju8c%2BffrivTwIkazx%2FauJ%2FgNzOJWtT6FJagEkFYiOPFoHpe%2Fd%2FXacgF9%2FEpA6gmoYeKIMxEaal9SPXT7YmOUylYt5ib7mJV6pmtKoSJz7YVksnIUrqNY4vhFqUH9mHH8FVSBaahyD5VtgGbEPA07W4EKwvBqkI9usqnFrDvyMwdFqIB8YTzGa%2BV3PXyg%2BbWhDqX80hd1mOQ2tYXuYy8xGOuzSZaDPnZf7N19JS8a8b%2BeeX7tyes5NXv%2BnngWNjdnZfaV2CADE1rR2y%2FoGd6kQdY6akuylxgOqrZAc9dmjBQzoZzhSzsWF07mSctgK2yygWd9iQsNp3sYZX65A01OFua%2BjPi7RN%2FoQpNeMNT8qJEGOqYBtbVn%2F9qtb2oaAQCFCScPYA8oBPDduCv%2FqRNABDMrsUQlRgUWrD%2FIdKr3Jgfck%2BwZQQo0ZqcQ0lgoKdDeJdumD2wllUDelGHC5aV9kM7YdDLR8hkxYXBLgaxxXuB7dRcAh%2BrAv8btu3PPNAuNre3Lk0Dyudahd2Np%2Fhbhtd0ZltbH1dVQaX6cFRsn3zV7sYIwwl2X8m0GOQgeDC5PdfQCDFWXvFf4cg%3D%3D&Expires=1647066887 [following]\n",
            "--2022-03-10 18:34:47--  https://encode-public.s3.amazonaws.com/2016/05/30/e3880aa9-e9ba-4114-8ba1-f1cf981147f4/ENCFF524OKJ.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF524OKJ.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX67YIJFZR&Signature=j16PNxjyITC%2FHr3WoGJlmxXRv18%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIr%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJGMEQCICzwbKESX2SuXYyDX%2BdCgEHD%2FB9f5AxE8lB0i7ojli1rAiAbrLuIUIolnIbd1%2FhJx1hGi8cKq1ec8V1RA6GQJXqJaSqDBAjz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMKc9cz76b1MoXw2abKtcDIt1PTzDTZ2cd41%2F6yhBhw7VoZnEJiAxRS3BgVKyIp3mesBtTfMdnhzWtTFz0AHyz9%2BMKJNj3G3fynFEi4TDQTCBqNDR6ag4ttaXiSgPDxfKgvvnQ5fxkx%2FJCt4hP%2FWCxiRkAFcM5amFuaA%2FNxlCpI5oy0yEGpP9o4K3%2B8vKdEl%2FPTrDUn0qFe%2FcDsvS%2Bh7g6aZr5c0Y8O9AsX51xppS8kE9Qma1DAZ02RamsO6DPs8LpZQXHoffZM7idcWjJZlR21mhVTBJLVTDxiYjeju8c%2BffrivTwIkazx%2FauJ%2FgNzOJWtT6FJagEkFYiOPFoHpe%2Fd%2FXacgF9%2FEpA6gmoYeKIMxEaal9SPXT7YmOUylYt5ib7mJV6pmtKoSJz7YVksnIUrqNY4vhFqUH9mHH8FVSBaahyD5VtgGbEPA07W4EKwvBqkI9usqnFrDvyMwdFqIB8YTzGa%2BV3PXyg%2BbWhDqX80hd1mOQ2tYXuYy8xGOuzSZaDPnZf7N19JS8a8b%2BeeX7tyes5NXv%2BnngWNjdnZfaV2CADE1rR2y%2FoGd6kQdY6akuylxgOqrZAc9dmjBQzoZzhSzsWF07mSctgK2yygWd9iQsNp3sYZX65A01OFua%2BjPi7RN%2FoQpNeMNT8qJEGOqYBtbVn%2F9qtb2oaAQCFCScPYA8oBPDduCv%2FqRNABDMrsUQlRgUWrD%2FIdKr3Jgfck%2BwZQQo0ZqcQ0lgoKdDeJdumD2wllUDelGHC5aV9kM7YdDLR8hkxYXBLgaxxXuB7dRcAh%2BrAv8btu3PPNAuNre3Lk0Dyudahd2Np%2Fhbhtd0ZltbH1dVQaX6cFRsn3zV7sYIwwl2X8m0GOQgeDC5PdfQCDFWXvFf4cg%3D%3D&Expires=1647066887\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.137.91\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.137.91|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 804989873 (768M) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF524OKJ.fastq.gz’\n",
            "\n",
            "ENCFF524OKJ.fastq.g 100%[===================>] 767.70M  42.5MB/s    in 19s     \n",
            "\n",
            "2022-03-10 18:35:06 (40.7 MB/s) - ‘ENCFF524OKJ.fastq.gz’ saved [804989873/804989873]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "# ChIP-seq на гистоновой метке\n",
        "!wget https://www.encodeproject.org/files/ENCFF891GXF/@@download/ENCFF891GXF.fastq.gz\n",
        "!wget https://www.encodeproject.org/files/ENCFF170YTR/@@download/ENCFF170YTR.fastq.gz\n",
        "# ChIP-seq контроль (SVK)\n",
        "!wget https://www.encodeproject.org/files/ENCFF524OKJ/@@download/ENCFF524OKJ.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "id": "v73fH7WiVDu9"
      },
      "outputs": [],
      "source": [
        "# распаковка ахривов\n",
        "!gzip -d /content/ENCFF891GXF.fastq.gz /content/ENCFF170YTR.fastq.gz /content/ENCFF524OKJ.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "6sPcIefNHx-L"
      },
      "source": [
        "## FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "5XM6jBcuGjMV",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "d784be38-95fe-448b-c0c6-72a4d606b501"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF891GXF.fastq\n",
            "Approx 5% complete for ENCFF891GXF.fastq\n",
            "Approx 10% complete for ENCFF891GXF.fastq\n",
            "Approx 15% complete for ENCFF891GXF.fastq\n",
            "Approx 20% complete for ENCFF891GXF.fastq\n",
            "Approx 25% complete for ENCFF891GXF.fastq\n",
            "Approx 30% complete for ENCFF891GXF.fastq\n",
            "Approx 35% complete for ENCFF891GXF.fastq\n",
            "Approx 40% complete for ENCFF891GXF.fastq\n",
            "Approx 45% complete for ENCFF891GXF.fastq\n",
            "Approx 50% complete for ENCFF891GXF.fastq\n",
            "Approx 55% complete for ENCFF891GXF.fastq\n",
            "Approx 60% complete for ENCFF891GXF.fastq\n",
            "Approx 65% complete for ENCFF891GXF.fastq\n",
            "Approx 70% complete for ENCFF891GXF.fastq\n",
            "Approx 75% complete for ENCFF891GXF.fastq\n",
            "Approx 80% complete for ENCFF891GXF.fastq\n",
            "Approx 85% complete for ENCFF891GXF.fastq\n",
            "Approx 90% complete for ENCFF891GXF.fastq\n",
            "Approx 95% complete for ENCFF891GXF.fastq\n",
            "Analysis complete for ENCFF891GXF.fastq\n",
            "Started analysis of ENCFF170YTR.fastq\n",
            "Approx 5% complete for ENCFF170YTR.fastq\n",
            "Approx 10% complete for ENCFF170YTR.fastq\n",
            "Approx 15% complete for ENCFF170YTR.fastq\n",
            "Approx 20% complete for ENCFF170YTR.fastq\n",
            "Approx 25% complete for ENCFF170YTR.fastq\n",
            "Approx 30% complete for ENCFF170YTR.fastq\n",
            "Approx 35% complete for ENCFF170YTR.fastq\n",
            "Approx 40% complete for ENCFF170YTR.fastq\n",
            "Approx 45% complete for ENCFF170YTR.fastq\n",
            "Approx 50% complete for ENCFF170YTR.fastq\n",
            "Approx 55% complete for ENCFF170YTR.fastq\n",
            "Approx 60% complete for ENCFF170YTR.fastq\n",
            "Approx 65% complete for ENCFF170YTR.fastq\n",
            "Approx 70% complete for ENCFF170YTR.fastq\n",
            "Approx 75% complete for ENCFF170YTR.fastq\n",
            "Approx 80% complete for ENCFF170YTR.fastq\n",
            "Approx 85% complete for ENCFF170YTR.fastq\n",
            "Approx 90% complete for ENCFF170YTR.fastq\n",
            "Approx 95% complete for ENCFF170YTR.fastq\n",
            "Analysis complete for ENCFF170YTR.fastq\n",
            "Started analysis of ENCFF524OKJ.fastq\n",
            "Approx 5% complete for ENCFF524OKJ.fastq\n",
            "Approx 10% complete for ENCFF524OKJ.fastq\n",
            "Approx 15% complete for ENCFF524OKJ.fastq\n",
            "Approx 20% complete for ENCFF524OKJ.fastq\n",
            "Approx 25% complete for ENCFF524OKJ.fastq\n",
            "Approx 30% complete for ENCFF524OKJ.fastq\n",
            "Approx 35% complete for ENCFF524OKJ.fastq\n",
            "Approx 40% complete for ENCFF524OKJ.fastq\n",
            "Approx 45% complete for ENCFF524OKJ.fastq\n",
            "Approx 50% complete for ENCFF524OKJ.fastq\n",
            "Approx 55% complete for ENCFF524OKJ.fastq\n",
            "Approx 60% complete for ENCFF524OKJ.fastq\n",
            "Approx 65% complete for ENCFF524OKJ.fastq\n",
            "Approx 70% complete for ENCFF524OKJ.fastq\n",
            "Approx 75% complete for ENCFF524OKJ.fastq\n",
            "Approx 80% complete for ENCFF524OKJ.fastq\n",
            "Approx 85% complete for ENCFF524OKJ.fastq\n",
            "Approx 90% complete for ENCFF524OKJ.fastq\n",
            "Approx 95% complete for ENCFF524OKJ.fastq\n",
            "Analysis complete for ENCFF524OKJ.fastq\n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc /content/ENCFF891GXF.fastq\n",
        "!./FastQC/fastqc /content/ENCFF170YTR.fastq\n",
        "!./FastQC/fastqc /content/ENCFF524OKJ.fastq # fastQC done"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! git clone https://github.com/t0pcup/hse_hw2_chip.git"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "gcPKuiTqJt-R",
        "outputId": "e1cf1271-5905-4333-8f8a-3576b21c10cf"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'hse_hw2_chip'...\n",
            "remote: Enumerating objects: 33, done.\u001b[K\n",
            "remote: Counting objects: 100% (33/33), done.\u001b[K\n",
            "remote: Compressing objects: 100% (22/22), done.\u001b[K\n",
            "remote: Total 33 (delta 11), reused 23 (delta 7), pack-reused 0\u001b[K\n",
            "Unpacking objects: 100% (33/33), done.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-TO3DQaUoGhB"
      },
      "source": [
        "### Trimmed FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Qellza2PI7fT",
        "outputId": "c6965af9-e203-423d-af51-38d245b12615"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "TrimmomaticSE: Started with arguments:\n",
            " -phred33 ENCFF891GXF.fastq ENCFF891GXF_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
            "Automatically using 2 threads\n",
            "java.io.FileNotFoundException: /content/TruSeq3-SE (No such file or directory)\n",
            "\tat java.io.FileInputStream.open0(Native Method)\n",
            "\tat java.io.FileInputStream.open(FileInputStream.java:195)\n",
            "\tat java.io.FileInputStream.<init>(FileInputStream.java:138)\n",
            "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
            "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
            "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 1019, in wait\n",
            "    return self._wait(timeout=timeout)\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 1653, in _wait\n",
            "    (pid, sts) = self._try_wait(0)\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 1611, in _try_wait\n",
            "    (pid, sts) = os.waitpid(self.pid, wait_flags)\n",
            "KeyboardInterrupt\n",
            "\n",
            "During handling of the above exception, another exception occurred:\n",
            "\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/bin/trimmomatic\", line 103, in <module>\n",
            "    main()\n",
            "  File \"/usr/local/bin/trimmomatic\", line 99, in main\n",
            "    sys.exit(subprocess.call(java_args))\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 341, in call\n",
            "    return p.wait(timeout=timeout)\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 1032, in wait\n",
            "    self._wait(timeout=sigint_timeout)\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 1647, in _wait\n",
            "    time.sleep(delay)\n",
            "KeyboardInterrupt\n",
            "/bin/bash: ./FastQC/fastqc: No such file or directory\n",
            "TrimmomaticSE: Started with arguments:\n",
            " -phred33 ENCFF170YTR.fastq ENCFF170YTR_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
            "Automatically using 2 threads\n",
            "java.io.FileNotFoundException: /content/TruSeq3-SE (No such file or directory)\n",
            "\tat java.io.FileInputStream.open0(Native Method)\n",
            "\tat java.io.FileInputStream.open(FileInputStream.java:195)\n",
            "\tat java.io.FileInputStream.<init>(FileInputStream.java:138)\n",
            "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
            "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
            "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 1019, in wait\n",
            "    return self._wait(timeout=timeout)\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 1653, in _wait\n",
            "    (pid, sts) = self._try_wait(0)\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 1611, in _try_wait\n",
            "    (pid, sts) = os.waitpid(self.pid, wait_flags)\n",
            "KeyboardInterrupt\n",
            "\n",
            "During handling of the above exception, another exception occurred:\n",
            "\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/bin/trimmomatic\", line 103, in <module>\n",
            "    main()\n",
            "  File \"/usr/local/bin/trimmomatic\", line 99, in main\n",
            "    sys.exit(subprocess.call(java_args))\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 341, in call\n",
            "    return p.wait(timeout=timeout)\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 1032, in wait\n",
            "    self._wait(timeout=sigint_timeout)\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 1647, in _wait\n",
            "    time.sleep(delay)\n",
            "KeyboardInterrupt\n",
            "/bin/bash: ./FastQC/fastqc: No such file or directory\n",
            "TrimmomaticSE: Started with arguments:\n",
            " -phred33 ENCFF524OKJ.fastq ENCFF524OKJ_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
            "Automatically using 2 threads\n",
            "java.io.FileNotFoundException: /content/TruSeq3-SE (No such file or directory)\n",
            "\tat java.io.FileInputStream.open0(Native Method)\n",
            "\tat java.io.FileInputStream.open(FileInputStream.java:195)\n",
            "\tat java.io.FileInputStream.<init>(FileInputStream.java:138)\n",
            "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
            "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
            "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 1019, in wait\n",
            "    return self._wait(timeout=timeout)\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 1653, in _wait\n",
            "    (pid, sts) = self._try_wait(0)\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 1611, in _try_wait\n",
            "    (pid, sts) = os.waitpid(self.pid, wait_flags)\n",
            "KeyboardInterrupt\n",
            "\n",
            "During handling of the above exception, another exception occurred:\n",
            "\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/bin/trimmomatic\", line 103, in <module>\n",
            "    main()\n",
            "  File \"/usr/local/bin/trimmomatic\", line 99, in main\n",
            "    sys.exit(subprocess.call(java_args))\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 341, in call\n",
            "    return p.wait(timeout=timeout)\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 1032, in wait\n",
            "    self._wait(timeout=sigint_timeout)\n",
            "  File \"/usr/local/lib/python3.7/subprocess.py\", line 1647, in _wait\n",
            "    time.sleep(delay)\n",
            "KeyboardInterrupt\n",
            "/bin/bash: ./FastQC/fastqc: No such file or directory\n"
          ]
        }
      ],
      "source": [
        "!trimmomatic SE -phred33 ENCFF891GXF.fastq ENCFF891GXF_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
        "!trimmomatic SE -phred33 ENCFF170YTR.fastq ENCFF170YTR_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
        "!trimmomatic SE -phred33 ENCFF524OKJ.fastq ENCFF524OKJ_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
        "\n",
        "!./FastQC/fastqc ENCFF891GXF_trimmed.fastq\n",
        "!./FastQC/fastqc ENCFF170YTR_trimmed.fastq\n",
        "!./FastQC/fastqc ENCFF524OKJ_trimmed.fastq"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "W9QHWBLlpMjQ"
      },
      "source": [
        "## Выравнивание на хромосому 13"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {
        "id": "N4JIGZOsGiY2",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "16268731-17c5-4a16-da8d-c31ef9eed143"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-10 19:00:16--  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr13.fa.gz\n",
            "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n",
            "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 31870393 (30M) [application/x-gzip]\n",
            "Saving to: ‘chr13.fa.gz’\n",
            "\n",
            "chr13.fa.gz         100%[===================>]  30.39M  27.2MB/s    in 1.1s    \n",
            "\n",
            "2022-03-10 19:00:18 (27.2 MB/s) - ‘chr13.fa.gz’ saved [31870393/31870393]\n",
            "\n",
            "Settings:\n",
            "  Output files: \"chromosome_index.*.bt2\"\n",
            "  Line rate: 6 (line is 64 bytes)\n",
            "  Lines per side: 1 (side is 64 bytes)\n",
            "  Offset rate: 4 (one in 16)\n",
            "  FTable chars: 10\n",
            "  Strings: unpacked\n",
            "  Max bucket size: default\n",
            "  Max bucket size, sqrt multiplier: default\n",
            "  Max bucket size, len divisor: 4\n",
            "  Difference-cover sample period: 1024\n",
            "  Endianness: little\n",
            "  Actual local endianness: little\n",
            "  Sanity checking: disabled\n",
            "  Assertions: disabled\n",
            "  Random seed: 0\n",
            "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
            "Input files DNA, FASTA:\n",
            "  chr13.fa\n",
            "Building a SMALL index\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:01\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:01\n",
            "bmax according to bmaxDivN setting: 24495781\n",
            "Using parameters --bmax 18371836 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 18371836 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:02\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:01\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 9.79831e+07 (target: 18371835)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 1\n",
            "  No samples; assembling all-inclusive block\n",
            "  Sorting block of length 97983125 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:01:01\n",
            "Returning block of 97983126 for bucket 1\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 30047611\n",
            "fchr[G]: 48886803\n",
            "fchr[T]: 67820408\n",
            "fchr[$]: 97983125\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 36855779 bytes to primary EBWT file: chromosome_index.1.bt2\n",
            "Wrote 24495788 bytes to secondary EBWT file: chromosome_index.2.bt2\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 97983125\n",
            "    bwtLen: 97983126\n",
            "    sz: 24495782\n",
            "    bwtSz: 24495782\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 6123946\n",
            "    offsSz: 24495784\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 510329\n",
            "    numLines: 510329\n",
            "    ebwtTotLen: 32661056\n",
            "    ebwtTotSz: 32661056\n",
            "    color: 0\n",
            "    reverse: 0\n",
            "Total time for call to driver() for forward index: 00:01:26\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:01\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:01\n",
            "  Time to reverse reference sequence: 00:00:00\n",
            "bmax according to bmaxDivN setting: 24495781\n",
            "Using parameters --bmax 18371836 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 18371836 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:02\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:01\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 9.79831e+07 (target: 18371835)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 1\n",
            "  No samples; assembling all-inclusive block\n",
            "  Sorting block of length 97983125 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:58\n",
            "Returning block of 97983126 for bucket 1\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 30047611\n",
            "fchr[G]: 48886803\n",
            "fchr[T]: 67820408\n",
            "fchr[$]: 97983125\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 36855779 bytes to primary EBWT file: chromosome_index.rev.1.bt2\n",
            "Wrote 24495788 bytes to secondary EBWT file: chromosome_index.rev.2.bt2\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 97983125\n",
            "    bwtLen: 97983126\n",
            "    sz: 24495782\n",
            "    bwtSz: 24495782\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 6123946\n",
            "    offsSz: 24495784\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 510329\n",
            "    numLines: 510329\n",
            "    ebwtTotLen: 32661056\n",
            "    ebwtTotSz: 32661056\n",
            "    color: 0\n",
            "    reverse: 1\n",
            "Total time for backward call to driver() for mirror index: 00:01:22\n"
          ]
        }
      ],
      "source": [
        "!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr13.fa.gz\n",
        "!gzip -d chr13.fa.gz\n",
        "\n",
        "!bowtie2-build chr13.fa chromosome_index"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "id": "E_usI8XUrQnR"
      },
      "outputs": [],
      "source": [
        "!mkdir bowtie2_res"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 18,
      "metadata": {
        "id": "3t80UATHqFe7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "cc2a248f-f600-4628-8b28-9105cb91a79c"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "43936063 reads; of these:\n",
            "  43936063 (100.00%) were unpaired; of these:\n",
            "    37341354 (84.99%) aligned 0 times\n",
            "    1554436 (3.54%) aligned exactly 1 time\n",
            "    5040273 (11.47%) aligned >1 times\n",
            "15.01% overall alignment rate\n",
            "CPU times: user 22.8 s, sys: 3.32 s, total: 26.1 s\n",
            "Wall time: 42min 15s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF891GXF.fastq \\\n",
        "          -S bowtie2_res/res_GXF.sam"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF170YTR.fastq \\\n",
        "          -S bowtie2_res/res_YTR.sam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "xIW-482WEnkA",
        "outputId": "48cb8275-433f-4d45-bef3-c6adc8f818d9"
      },
      "execution_count": 9,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "44100884 reads; of these:\n",
            "  44100884 (100.00%) were unpaired; of these:\n",
            "    38851211 (88.10%) aligned 0 times\n",
            "    1250566 (2.84%) aligned exactly 1 time\n",
            "    3999107 (9.07%) aligned >1 times\n",
            "11.90% overall alignment rate\n",
            "CPU times: user 16.9 s, sys: 2.07 s, total: 18.9 s\n",
            "Wall time: 32min 32s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF524OKJ.fastq \\\n",
        "          -S bowtie2_res/res_OKJ.sam"
      ],
      "metadata": {
        "id": "xlrA58UZ6Qd_",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "6b3f76cc-7c91-4d5f-ffad-2a26f9281701"
      },
      "execution_count": 10,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "27267571 reads; of these:\n",
            "  27267571 (100.00%) were unpaired; of these:\n",
            "    23368148 (85.70%) aligned 0 times\n",
            "    971508 (3.56%) aligned exactly 1 time\n",
            "    2927915 (10.74%) aligned >1 times\n",
            "14.30% overall alignment rate\n",
            "CPU times: user 12 s, sys: 1.48 s, total: 13.5 s\n",
            "Wall time: 22min 14s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XZtQrPZMmgnC"
      },
      "source": [
        "## Peak calling\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {
        "id": "uX1ps4VDp_Ae"
      },
      "outputs": [],
      "source": [
        "! mkdir macs2_1\n",
        "! mkdir macs2_2"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_GXF.sam \\\n",
        "    -c bowtie2_res/res_OKJ.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2_1"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3kk8UGVGFoks",
        "outputId": "b8d49c97-a6f7-4bfd-e0cc-5e22f4177d7d"
      },
      "execution_count": 19,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Thu, 10 Mar 2022 20:46:07: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_GXF.sam -c bowtie2_res/res_OKJ.sam -f SAM --outdir macs2_1\n",
            "# ARGUMENTS LIST:\n",
            "# name = NA\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_GXF.sam']\n",
            "# control file = ['bowtie2_res/res_OKJ.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Thu, 10 Mar 2022 20:46:07: #1 read tag files... \n",
            "INFO  @ Thu, 10 Mar 2022 20:46:07: #1 read treatment tags... \n",
            "INFO  @ Thu, 10 Mar 2022 20:46:09:  2000000 \n",
            "INFO  @ Thu, 10 Mar 2022 20:46:12:  4000000 \n",
            "INFO  @ Thu, 10 Mar 2022 20:46:22:  13000000 \n",
            "INFO  @ Thu, 10 Mar 2022 20:46:30:  20000000 \n",
            "INFO  @ Thu, 10 Mar 2022 20:46:34:  23000000 \n",
            "INFO  @ Thu, 10 Mar 2022 20:46:48:  35000000 \n",
            "INFO  @ Thu, 10 Mar 2022 20:46:57:  43000000 \n",
            "INFO  @ Thu, 10 Mar 2022 20:46:59: #1.2 read input tags... \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:17:  5000000 \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:26:  10000000 \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:34:  14000000 \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:45:  20000000 \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:56:  27000000 \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:57: #1 tag size is determined as 51 bps \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:57: #1 tag size = 51.0 \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:57: #1  total tags in treatment: 6594709 \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:57: #1 user defined the maximum tags... \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:57: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:57: #1  tags after filtering in treatment: 5101916 \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:57: #1  Redundant rate of treatment: 0.23 \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:57: #1  total tags in control: 3899423 \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:57: #1 user defined the maximum tags... \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:57: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:57: #1  tags after filtering in control: 3242657 \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:57: #1  Redundant rate of control: 0.17 \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:57: #1 finished! \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:57: #2 Build Peak Model... \n",
            "INFO  @ Thu, 10 Mar 2022 20:47:57: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Thu, 10 Mar 2022 20:48:00: #2 number of paired peaks: 40200 \n",
            "INFO  @ Thu, 10 Mar 2022 20:48:00: start model_add_line... \n",
            "INFO  @ Thu, 10 Mar 2022 20:48:00: start X-correlation... \n",
            "INFO  @ Thu, 10 Mar 2022 20:48:00: end of X-cor \n",
            "INFO  @ Thu, 10 Mar 2022 20:48:00: #2 finished! \n",
            "INFO  @ Thu, 10 Mar 2022 20:48:00: #2 predicted fragment length is 50 bps \n",
            "INFO  @ Thu, 10 Mar 2022 20:48:00: #2 alternative fragment length(s) may be 50,396 bps \n",
            "INFO  @ Thu, 10 Mar 2022 20:48:00: #2.2 Generate R script for model : macs2_1/NA_model.r \n",
            "WARNING @ Thu, 10 Mar 2022 20:48:00: #2 Since the d (50) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Thu, 10 Mar 2022 20:48:00: #2 You may need to consider one of the other alternative d(s): 50,396 \n",
            "WARNING @ Thu, 10 Mar 2022 20:48:00: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Thu, 10 Mar 2022 20:48:00: #3 Call peaks... \n",
            "INFO  @ Thu, 10 Mar 2022 20:48:00: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Thu, 10 Mar 2022 20:48:00: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Thu, 10 Mar 2022 20:48:20: #3 Call peaks for each chromosome... \n",
            "INFO  @ Thu, 10 Mar 2022 20:48:31: #4 Write output xls file... macs2_1/NA_peaks.xls \n",
            "INFO  @ Thu, 10 Mar 2022 20:48:31: #4 Write broad peak in broadPeak format file... macs2_1/NA_peaks.broadPeak \n",
            "INFO  @ Thu, 10 Mar 2022 20:48:31: #4 Write broad peak in bed12/gappedPeak format file... macs2_1/NA_peaks.gappedPeak \n",
            "INFO  @ Thu, 10 Mar 2022 20:48:31: Done! \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_YTR.sam \\\n",
        "    -c bowtie2_res/res_OKJ.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2_2"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zvqsNJYOJbkS",
        "outputId": "69518c58-04ae-4f3a-eed0-fbb0b7c7cde6"
      },
      "execution_count": 12,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Thu, 10 Mar 2022 19:57:56: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_YTR.sam -c bowtie2_res/res_OKJ.sam -f SAM --outdir macs2_2\n",
            "# ARGUMENTS LIST:\n",
            "# name = NA\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_YTR.sam']\n",
            "# control file = ['bowtie2_res/res_OKJ.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Thu, 10 Mar 2022 19:57:56: #1 read tag files... \n",
            "INFO  @ Thu, 10 Mar 2022 19:57:56: #1 read treatment tags... \n",
            "INFO  @ Thu, 10 Mar 2022 19:58:00:  3000000 \n",
            "INFO  @ Thu, 10 Mar 2022 19:58:10:  11000000 \n",
            "INFO  @ Thu, 10 Mar 2022 19:58:14:  15000000 \n",
            "INFO  @ Thu, 10 Mar 2022 19:58:26:  26000000 \n",
            "INFO  @ Thu, 10 Mar 2022 19:58:34:  33000000 \n",
            "INFO  @ Thu, 10 Mar 2022 19:58:36:  35000000 \n",
            "INFO  @ Thu, 10 Mar 2022 19:58:40:  38000000 \n",
            "INFO  @ Thu, 10 Mar 2022 19:58:47: #1.2 read input tags... \n",
            "INFO  @ Thu, 10 Mar 2022 19:58:52:  5000000 \n",
            "INFO  @ Thu, 10 Mar 2022 19:58:58:  10000000 \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:02:  14000000 \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:09:  20000000 \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:17:  27000000 \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:18: #1 tag size is determined as 51 bps \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:18: #1 tag size = 51.0 \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:18: #1  total tags in treatment: 5249673 \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:18: #1 user defined the maximum tags... \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:18: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:18: #1  tags after filtering in treatment: 4110822 \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:18: #1  Redundant rate of treatment: 0.22 \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:18: #1  total tags in control: 3899423 \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:18: #1 user defined the maximum tags... \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:18: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:18: #1  tags after filtering in control: 3242657 \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:18: #1  Redundant rate of control: 0.17 \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:18: #1 finished! \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:18: #2 Build Peak Model... \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:18: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:19: #2 number of paired peaks: 34568 \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:19: start model_add_line... \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:19: start X-correlation... \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:19: end of X-cor \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:19: #2 finished! \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:19: #2 predicted fragment length is 50 bps \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:19: #2 alternative fragment length(s) may be 50 bps \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:19: #2.2 Generate R script for model : macs2_2/NA_model.r \n",
            "WARNING @ Thu, 10 Mar 2022 19:59:19: #2 Since the d (50) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Thu, 10 Mar 2022 19:59:19: #2 You may need to consider one of the other alternative d(s): 50 \n",
            "WARNING @ Thu, 10 Mar 2022 19:59:19: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:19: #3 Call peaks... \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:19: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:19: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:38: #3 Call peaks for each chromosome... \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:49: #4 Write output xls file... macs2_2/NA_peaks.xls \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:49: #4 Write broad peak in broadPeak format file... macs2_2/NA_peaks.broadPeak \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:49: #4 Write broad peak in bed12/gappedPeak format file... macs2_2/NA_peaks.gappedPeak \n",
            "INFO  @ Thu, 10 Mar 2022 19:59:49: Done! \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {
        "id": "HrNwxs7ArPz7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "3d10eddc-8d17-45ac-ee20-a3946195f6f4"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-10 19:59:50--  https://www.encodeproject.org/files/ENCFF573MUH/@@download/ENCFF573MUH.bed.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2020/10/01/aace6c98-55f9-4b34-bba0-30359fcae5fc/ENCFF573MUH.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF573MUH.bed.gz&AWSAccessKeyId=ASIATGZNGCNXW7DO57OE&Signature=EdFUmw6Ia4wzK9apPw4QFPo28LY%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIv%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIQDXgNPZ7hqP51imUatkweYAmWCe9py5ZRJvfOf4%2FUB9VwIgEphBmMlPWKsqLIzX7JjRRUJRPNcHZRRNJBQMAeS1XlAqgwQI9P%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDOZCBbs2V9VYpbeT0SrXAzswUh7FHTAGdFaExBOIAbLxkos9BIWMwkl4NBE%2FxiIUfIvaNpsA96aToPqizJaceM7vsmzlxF7LmbBOXonvYD%2BqVjR5rIrySfQJy4G9QWI1I3FKRqTU0Gg2xFODiz5IkcIge3u%2FFlD6JIDiQw0tHp34gKWSLqpDQ5MXY6BjdG5aCBsRJLkU98uUF4Pr%2BrHay1WxtqeVkSHfQzG2Xz6AU1%2BGzD0nyv%2F6yiDNFMCgQzlkXYq1V6NmoPRy1kRnJl8rm0HR1TbWMVt%2FuiAwWuucErqPpUDh%2B6Y3VHul2%2BHpAYuaJQGmC23RmgE%2BJ2mcWvXXZP1bffxvGRbdUlg0a7FEd1uIYAiID6GElFDdrNgEF32rQrpAf2y25%2FW5KmCli9224FIGxS%2BWOa6J%2Bs8A8wo8xW7PrHK1jLm1hiWvmSD6%2B0YKKLCWG%2Frg6OWS%2FiM6f1t4P8Y5MV5%2FWj2A6YcIcNNnDGl7Yp23GogI8GSh0MCXpH4DKHNYQmJ%2B4pBwqPB6yH2wjMq3Pw1H5aTdDcf0HrjCHQSXeo%2FLUaxhL%2BeVPtiB8CrnAE66wQmbS8vn71VKe4jasjHF1fgLkNHAQbBobaaycJtZt4nJo3oDGvcQZ%2F6phC3L0wROI9mXtjCnmqmRBjqlASYN9%2FK9zISNmeMdC4PoyVTH%2FWbAuRzG%2BF8AFMrnX6xoxHL5R4TjZ6a11EFOJy%2BYYTMxmOWH2LICwMO%2FLAZAdPGSnFUZ7%2F3U4I9SoQVjIWRLflulmhDma0XKIAeQqzlHlmNF75kd%2Bxcs%2BOrgRnofKuBeom0Hll15sS0BvwIFUPaFNDlUMgce5KK0bJSXUh5PVt7guYwygqGC7OUIbZvV49IZ5qzrIw%3D%3D&Expires=1647071990 [following]\n",
            "--2022-03-10 19:59:50--  https://encode-public.s3.amazonaws.com/2020/10/01/aace6c98-55f9-4b34-bba0-30359fcae5fc/ENCFF573MUH.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF573MUH.bed.gz&AWSAccessKeyId=ASIATGZNGCNXW7DO57OE&Signature=EdFUmw6Ia4wzK9apPw4QFPo28LY%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIv%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIQDXgNPZ7hqP51imUatkweYAmWCe9py5ZRJvfOf4%2FUB9VwIgEphBmMlPWKsqLIzX7JjRRUJRPNcHZRRNJBQMAeS1XlAqgwQI9P%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDOZCBbs2V9VYpbeT0SrXAzswUh7FHTAGdFaExBOIAbLxkos9BIWMwkl4NBE%2FxiIUfIvaNpsA96aToPqizJaceM7vsmzlxF7LmbBOXonvYD%2BqVjR5rIrySfQJy4G9QWI1I3FKRqTU0Gg2xFODiz5IkcIge3u%2FFlD6JIDiQw0tHp34gKWSLqpDQ5MXY6BjdG5aCBsRJLkU98uUF4Pr%2BrHay1WxtqeVkSHfQzG2Xz6AU1%2BGzD0nyv%2F6yiDNFMCgQzlkXYq1V6NmoPRy1kRnJl8rm0HR1TbWMVt%2FuiAwWuucErqPpUDh%2B6Y3VHul2%2BHpAYuaJQGmC23RmgE%2BJ2mcWvXXZP1bffxvGRbdUlg0a7FEd1uIYAiID6GElFDdrNgEF32rQrpAf2y25%2FW5KmCli9224FIGxS%2BWOa6J%2Bs8A8wo8xW7PrHK1jLm1hiWvmSD6%2B0YKKLCWG%2Frg6OWS%2FiM6f1t4P8Y5MV5%2FWj2A6YcIcNNnDGl7Yp23GogI8GSh0MCXpH4DKHNYQmJ%2B4pBwqPB6yH2wjMq3Pw1H5aTdDcf0HrjCHQSXeo%2FLUaxhL%2BeVPtiB8CrnAE66wQmbS8vn71VKe4jasjHF1fgLkNHAQbBobaaycJtZt4nJo3oDGvcQZ%2F6phC3L0wROI9mXtjCnmqmRBjqlASYN9%2FK9zISNmeMdC4PoyVTH%2FWbAuRzG%2BF8AFMrnX6xoxHL5R4TjZ6a11EFOJy%2BYYTMxmOWH2LICwMO%2FLAZAdPGSnFUZ7%2F3U4I9SoQVjIWRLflulmhDma0XKIAeQqzlHlmNF75kd%2Bxcs%2BOrgRnofKuBeom0Hll15sS0BvwIFUPaFNDlUMgce5KK0bJSXUh5PVt7guYwygqGC7OUIbZvV49IZ5qzrIw%3D%3D&Expires=1647071990\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.92.194.249\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.92.194.249|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 730641 (714K) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF573MUH.bed.gz’\n",
            "\n",
            "ENCFF573MUH.bed.gz  100%[===================>] 713.52K  2.26MB/s    in 0.3s    \n",
            "\n",
            "2022-03-10 19:59:51 (2.26 MB/s) - ‘ENCFF573MUH.bed.gz’ saved [730641/730641]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF573MUH/@@download/ENCFF573MUH.bed.gz\n",
        "!gzip -d ENCFF573MUH.bed.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "MYEGEbQjvEwY"
      },
      "source": [
        "## Сравнение результатов\n",
        "\n",
        "Сравниваем те пики, которые мы получили, с пиками, которые приведены в ENCODE (важно, чтобы версии генома hg38 или hg19) для .bed файла из ENCODE и той хромосомы, которую скачивали выше, сопадали.\n",
        "\n",
        "Проанализируйте полученные результаты и приведите свои рассуждения в README.md. Как можно объяснить различия в количестве пересечений?"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!intervene venn -i macs2_1/NA_peaks.broadPeak ENCFF573MUH.bed --filenames \\\n",
        "    --output venn_results/venn1_GXF\n",
        "!intervene venn -i ENCFF573MUH.bed macs2_1/NA_peaks.broadPeak --filenames \\\n",
        "    --output venn_results/venn2_GXF"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "seb1pb6jOZ-9",
        "outputId": "86a97ce7-5e42-4a2a-f10d-f3fc8fd03264"
      },
      "execution_count": 20,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn1_GXF. \n",
            "Thank you for using Intervene!\n",
            "\n",
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn2_GXF. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!intervene venn -i macs2_2/NA_peaks.broadPeak ENCFF573MUH.bed --filenames \\\n",
        "    --output venn_results/venn1_YTR\n",
        "!intervene venn -i ENCFF573MUH.bed macs2_2/NA_peaks.broadPeak --filenames \\\n",
        "    --output venn_results/venn2_YTR"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZhhWDi7yOa2z",
        "outputId": "712a973d-78bc-4daa-d64a-dab2e006c3b0"
      },
      "execution_count": 14,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn1_YTR.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n",
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn2_YTR.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ]
    }
  ],
  "metadata": {
    "colab": {
      "collapsed_sections": [
        "w_TpdMixmsdY"
      ],
      "name": "hw2-chipseq",
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    },
    "accelerator": "TPU"
  },
  "nbformat": 4,
  "nbformat_minor": 0
}